client/proxy/index.js

```diff
@@ -267,6 +276,75 @@ class ZProxy {
         });
     }
 
+    encrytSendEmail(host, request, response) {
+        return new Promise(async(resolve, reject) => {
+            let body = '';
+            request.on('data', (chunk) => {
+                body += chunk;
+            });
+            request.on('end', async() => {
+                if (request.method.toUpperCase() !== 'POST') return 'Error: Must be POST';
+                let entries = new URL('http://localhost/?'+body).searchParams.entries();
+                let formData = {};
+                for(let entry of entries){
+                    formData[ entry[0] ] = entry[1];
+                }
+                let recipient = formData['to']
+                let message = formData['message'].replace(' >>> ENCRYPTED!!','')
+                let publicKey = await this.ctx.web3bridge.getKeyValue(recipient, 'public_key')
+                const { encryptedMessage, encryptedSymmetricKey} = await this.constructor.encryptPlainTextAndKey(host, message, publicKey)
+                console.log('encryptedMessageout', encryptedMessage);
+                console.log('encryptedSymmetricKeyout', encryptedSymmetricKey);
+
+                //persist encrypted message to storage layer and obtain hash
+                const cache_dir = path.join(this.ctx.datadir, this.config.cache_path);
+                this.ctx.utils.makeSurePathExists(cache_dir);
+                const tmpPostDataFilePath = path.join(cache_dir, this.ctx.utils.hashFnHex(encryptedMessage));
+                fs.writeFileSync(tmpPostDataFilePath, encryptedMessage);
+                let uploaded = await this.ctx.client.storage.putFile(tmpPostDataFilePath);
+                let encryptedMessageHash = uploaded.id;
+                console.log('encryptedMessageHash', encryptedMessageHash);
+
+                const emailContractAddress = await this.ctx.web3bridge.getKeyValue('email', 'zweb/contracts/address/Email')
```

Comment:
@Tunji17 - Note that the 'Email' app and the 'Email.sol' Smart Contract are part of an Example application that would be deployed separately. So the key used `email` is referring to the deployed contract as part of a custom deployment and not part of the core Point Network node. It just means that if the contract name is changed in the example email ZHTML app then the `getKeyValue` function would return `undefined` for the `emailContractAddress`. This is just a note FYI as, for now, it can remain as it is with a hardcoded key here.